specs/ExtendedNewWindowRequested.md

```diff
@@ -0,0 +1,109 @@
+# Background
+WebView2 raises the `NewWindowRequested` event when a new window is opened. The WebView2 team has been asked to include the name of the new window 
+being created as a parameter of `NewWindowRequestedEventArgs`. This window name is the name given to the window when it is opened 
+using `window.open(url, windowName)`.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose extending the `NewWindowRequestedEventArgs` to provide access to the `Name` property. 
+The `Name` property will return the name of the new window being created. 
+
+# Examples
+## C++: Get name of window
+
+``` cpp
+wil::com_ptr<ICoreWebView2> m_webviewEventSource;
+EventRegistrationToken m_newWindowRequestedToken = {};
+
+m_webviewEventSource->add_NewWindowRequested(
+    Callback<ICoreWebView2NewWindowRequestedEventHandler>(
+        [this](ICoreWebView2* sender, ICoreWebView2NewWindowRequestedEventArgs* args)
+          -> HRESULT 
+            {
+                wil::com_ptr<ICoreWebView2NewWindowRequestedEventArgs2> args2;
+              
+                if(SUCCEEDED(args->QueryInterface(IID_PPV_ARGS(&args2)))) 
+                {
+                    wil::unique_cotaskmem_string name;
+                    CHECK_FAILURE(args2->get_Name(&name));
+
+                    wil::unique_cotaskmem_string uri;
+                    CHECK_FAILURE(args->get_Uri(&uri));
+
+                    // Example usage of how the customer would use the window
+                    // name to handle special cases such as "openInNewBrowser"
+                    // which deviate from the normal handling of a new window.
+                    if (wcscmp(name.get(), L"openInNewBrowser") == 0)
+                    {
+                        ShellExecute(NULL, "open", uri.get(), 
+                            NULL, NULL, SW_SHOWNORMAL);
+                        args->put_Handled(true);
+                    }
+                    else 
+                    {
+                        HandleNormalNewWindow();
+                    }
+                }
+
+              return S_OK;
+            })
+            .Get(),
+  &m_newWindowRequestedToken);
+```
+
+## C#: Get name of window
+```c#
+webView.CoreWebView2.NewWindowRequested += WebView_NewWindowRequested;
+
+void WebView_NewWindowRequested(object sender, CoreWebView2NewWindowRequestedEventArgs e)
+{
+    string newWindowName = e.Name;
+    
+    // Example usage of how the customer would use the window
+    // name to handle special cases such as "openInNewBrowser"
+    // which deviate from the normal handling of a new window.
+    if (newWindowName == "openInNewBrowser")
+    {
+        Process.Start(e.Uri);
+        e.Handled = true;
+    }
+    else
+    {
+        HandleNormalNewWindow();
```

Comment:
Nit: we need to pass in event args to actually handle it. Similarly in the Win32 sample.
```suggestion
        HandleNewWindow(e); // App specific handling of opening a new window
```

Reply:
Added now for .NET and Win32 example